Makefile

```diff
@@ -58,10 +58,8 @@ type-check:
 
 .PHONY: doclinter
 doclinter:
-	@sphinx-lint --enable all --max-line-length 85 --disable triple-backticks \
-	             -i CHANGES.rst -i LICENSE.rst -i EXAMPLES.rst \
-	             $(addprefix -i doc/, _build _static _templates _themes) \
-	             *.rst doc/ 2>&1 | sort -V
+	@sphinx-lint --enable line-too-long --max-line-length 85 \
```

Comment:
```suggestion
	@sphinx-lint --enable all --max-line-length 85 \
```

There are only a few items we'd need to cleanup to remove `max-line-length`, too.

Reply:
- I want to have all lints enabled by default personally.
- I removed CHANGES.rst but I forgot to fix the `double backticks` errors. Note that without the `all` flag, the lint is NOT enabled so it's not good for me.

Reply:
So then are you suggesting we also change what is run on the CI? I'm fine with that

Obviously the main thing I want to achieve here is that, if `make check` passes locally, then there should be no reason for the CI linting to fail 😅

Reply:
Could we make the CI run `make doclinter` to reduce the chance of going out-of-sync again?

Reply:
> So then are you suggesting we also change what is run on the CI? I'm fine with that

Oh, I thought that the CI/CD was using our Makefile (or had the same rules). So yes, both the CI/CD and the local checks must be the same.

Reply:
For the CHANGES.rst, it's my bad. So, I suggest another PR to fix first the CHANGES.rst and then we update the CI/CD to use the same lint configuration.

Reply:
> Could we make the CI run make doclinter to reduce the chance of going out-of-sync again?

Seems reasonable to me 👍

Reply:
done